### PR TITLE
Table: Add initial row index 

### PR DESCRIPTION
**Pull request by gtk-grafana**
**What is this feature?**
Adding new `initialRowIndex` prop to <Table> component.
`initialRowIndex` sets the initial state of `rowHighlightIndex` in [RowsList.tsx](https://github.com/grafana/grafana/pull/82200/files#diff-94f0e6d49c1bbe3dff6f59ae4b08255fc40848b34e6a3d44fe6ffdafb5716b03R73)

image

**Why do we need this feature?**
Allows consumers of `<Table>` component to pass in row index that be highlighted and initially scrolled to.

**Who is this feature for?**
Grafana developers working with table component.

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: packages/grafana-ui/src/components/CustomScrollbar/CustomScrollbar.tsx [hunk 1]
```diff
@@ -64,6 +68,12 @@ export const CustomScrollbar = ({
     }
   }, [scrollTop]);
 
+  useEffect(() => {
+    if (ref.current && scrollToRowIndex != null && height) {
+      ref.current.scrollTop(scrollToRowIndex * height);
```
Comment (torkelo) [inline packages/grafana-ui/src/components/CustomScrollbar/CustomScrollbar.tsx:73]: Why not use scrollTop prop? this height* scrollToRowIndex seems like something that should not be inside CustomScrollbar

Comment (gtk-grafana) [reply packages/grafana-ui/src/components/CustomScrollbar/CustomScrollbar.tsx]: good catch, looks like we can simply set the initial state of `rowHighlightIndex` in the RowsList and all of this functionality is already built

File: packages/grafana-ui/src/components/Table/RowsList.tsx [hunk 2]
```diff
@@ -211,7 +213,6 @@ export const RowsList = (props: RowsListProps) => {
       if (rowHighlightIndex !== undefined && row.index === rowHighlightIndex) {
         style = { ...style, backgroundColor: theme.components.table.rowHoverBackground };
```
Comment (gtk-grafana) [inline packages/grafana-ui/src/components/Table/RowsList.tsx:214]:
```suggestion
        style = { ...style, backgroundColor: theme.colors.background.secondary };
```
Should we change the color so there's some differentiation between "hover" and "scrolledTo"?
image

Although maybe that isn't enough, can't tell which is which in light mode: 
image

Comment (gtk-grafana) [reply packages/grafana-ui/src/components/Table/RowsList.tsx:215]: Deal with this in another issue, don't increase the scope of this pr

**Review by codeincarnate — APPROVED** [review]
Tested out and worked just fine! Looks good 😄

**Pull request merged**